quantum/secure.c

```diff
@@ -85,3 +88,16 @@ void secure_task(void) {
     }
 #endif
 }
+
+#if defined(SECURE_ENABLE)
+__attribute__((weak)) bool secure_hook_user(secure_status_t secure_status) {
+    return true;
+}
+__attribute__((weak)) bool secure_hook_kb(secure_status_t secure_status) {
+    return secure_hook_user(secure_status);
+}
+
+__attribute__((weak)) void secure_hook_quantum(secure_status_t secure_status) {
+    secure_hook_kb(secure_status);
+}
```

Comment:
This still feels somewhat wrong, as it would contain the XAP activity hooks? Or if the plan is to add a non weak version in `quantum.h`, then we should add a comment here to ensure it doesnt get modified.

Though at this point it might be better if its reworked as part of XAP.

Reply:
Would it make sense/be best to `secure_hook_quantum` handle the clear, if transitioning to "pending", and have that in quantum.c? 

Eg, throw this in quantum.c (and only this)

```c
void secure_hook_quantum(secure_status_t secure_status) {
	if (secure_status == SECURE_PENDING) {
            clear_keyboard();
            layer_clear();
    }        

    secure_hook_kb(secure_status);
}
```
That would handle the keyboard clearing only when an unlock request is sent, and xap could be added here, as well.